lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/ClassHierarchyImpl.cs

```diff
@@ -167,6 +167,18 @@ public INode RegisterType(Type type)
                         RegisterType(constructorArg.Gettype());  // Gettype returns param's Type.fullname
                         if (constructorArg.GetNamedParameterName() != null)
                         {
+                            var pn = RegisterType(constructorArg.GetNamedParameterName());
+                            
+                            if (!(pn is INamedParameterNode))
+                            {
+                                string fullNamedParameterName = constructorArg.GetNamedParameterName();
```

Comment:
In the error message, {0} should be class B and {1} should class A. 
You can get class B from constructorArg.GetNamedParameterName(). It doesn't matter it is full name or short name. Full name should be fine and you don't need to get substring. 
For Class A, you get it from constructorDef.GetClassName(). Full name should be fine.